src/Sphere.jl

```diff
@@ -76,6 +76,27 @@ function zero_tangent_vector!(S::Sphere, v, x)
     return v
 end
 
+function vector_transport!(M::Sphere, vto, x, v, y)
```

Comment:
I wonder if a simpler interface is `vector_transport!(M, wto, x, v, w)`, which transports `w` from `x` to `exp(M, x, v)`. That would eliminate, for example, the `v_xy` computation below, since `vector_transport` will usually be coupled with `exp`, so a velocity will already exist.

Reply:
If so, I would like to have both interfaces, if possible (maybe only with types?).

Reply:
Hmm... I think these two functions (transporting in a direction and transporting to a point) should have different names. We'll also need vector transport along a curve. How about `vector_transport_point!`, `vector_transport_tvector!` and `vector_transport_curve!`?

Reply:
I don't mind having different names, but I would prefer names, that indicate _what_ we are doing instead of what you have to provide. What about

* `vector_transport_to!` for the point version, since we want to transport _to_ `p`
* `vector_transport_direction!` for the vector version, since the tangent vector indicates the direction? 
* `vector_transport_along!` for the curve version?

I am not that happy with the second name yet, but if we come up with good ones, we can also distinguish the to ways of specifying geodesics in the same naming scheme.

Reply:
> ...I would prefer names, that indicate _what_ we are doing instead of what you have to provide. What about
> 
> * `vector_transport_to!` for the point version, since we want to transport _to_ `p`
> * `vector_transport_direction!` for the vector version, since the tangent vector indicates the direction? 
> * `vector_transport_along!` for the curve version?
> 
> I am not that happy with the second name yet, but if we come up with good ones, we can also distinguish the to ways of specifying geodesics in the same naming scheme.
> 

So the base case here would be `vector_transport_along!`, which would require specification of a curve, either a function or a map (#28). `vector_transport_to` would call this with `shortest_geodesic`, while `vector_transport_direction` would call this with `geodesic`, right? I think `shortest_geodesic` and `geodesic` are adequately descriptive, but I don't see how we can adopt that terminology here. Perhaps `vector_transport_along_geodesic!` instead of `vector_transport_direction`. Likewise, we could do `vector_transport_along_shortest_geodesic!`, but that's just ugly. 

I guess the generalization of this would be `fiber_transport_XXX` or `transport_fiber` (I prefer this order since it's the fiber that is being transported, not the other way around). Alternatively, we can just provide `transport` and use types to specify what is being transported and how. e.g. `transport!(::TangentBundleFibers, xto, vto, x, v, curve)`.

Reply:
No, `vector_transport_along!`, is definitely not the base case. Usually it would be implemented by sampling the curve and calling `vector_transport_to!` many times. The base case would either be the `_to!` variant or the `_direction!` variant. I don't have a preference here, one has to do `log` and the other has to do `exp` anyway.

> I guess the generalization of this would be `fiber_transport_XXX` or `transport_fiber` (I prefer this order since it's the fiber that is being transported, not the other way around). Alternatively, we can just provide `transport` and use types to specify what is being transported and how. e.g. `transport!(::TangentBundleFibers, xto, vto, x, v, curve)`.

`transport_vector` is fine as a generalization, it just needs the same trick with `FVector` that I did with `flat!` and `sharp!`. By the way, fiber in this case refers to an entire vector space.

Reply:
I would also be fine with the last idea of just naming in `transport` and use dispatch; and yes, I think the default would be the point/`_to` variant.

Reply:
> No, `vector_transport_along!`, is definitely not the base case. Usually it would be implemented by sampling the curve and calling `vector_transport_to!` many times.

Oh, sorry, of course. I was thinking more from a math perspective than implementation. It's true that the curve variant would need something like Schild's ladder and couldn't be the default.